cli/command/image/tree.go

```diff
@@ -0,0 +1,296 @@
+}
+
+type imageDetails struct {
+	ID        string
+	DiskUsage string
+	Used      bool
+}
+
+type topImage struct {
+	Names    []string
+	Details  imageDetails
+	Children []subImage
+}
+
+type subImage struct {
+	Platform  string
+	Available bool
+	Details   imageDetails
+}
+
+const headerSpacing = 3
+
+//nolint:gocyclo
+func printImageTree(dockerCLI command.Cli, images []topImage) error {
+	out := dockerCLI.Out()
+	_, width := out.GetTtySize()
+	if width == 0 {
+		width = 80
+	}
+	if width < 20 {
+		width = 20
+	}
+
+	headers := []header{
+		{Title: "Image", Width: 0, Left: true},
+		{Title: "ID", Width: 12},
+		{Title: "Disk usage", Width: 10},
+		{Title: "Used", Width: 4},
+	}
+
+	nameWidth := int(width)
+	for idx, h := range headers {
+		if h.Width == 0 {
+			continue
+		}
+		d := h.Width + headerSpacing
+		// If the first column gets too short, remove remaining columns
+		if nameWidth-d < 12 {
+			headers = headers[:idx]
+			break
+		}
+		nameWidth -= d
+	}
+
+	maxImageName := len(headers[0].Title)
+	for _, img := range images {
+		for _, name := range img.Names {
+			if len(name) > maxImageName {
+				maxImageName = len(name)
+			}
+		}
+		for _, sub := range img.Children {
+			if len(sub.Platform) > maxImageName {
+				maxImageName = len(sub.Platform)
+			}
+		}
+	}
+
+	if nameWidth > maxImageName+headerSpacing {
+		nameWidth = maxImageName + headerSpacing
+	}
+
+	if nameWidth < 0 {
+		headers = headers[:1]
+		nameWidth = int(width)
+	}
+	headers[0].Width = nameWidth
+
+	headerColor := aec.NewBuilder(aec.DefaultF, aec.Bold).ANSI
+	topNameColor := aec.NewBuilder(aec.BlueF, aec.Underline, aec.Bold).ANSI
+	normalColor := aec.NewBuilder(aec.DefaultF).ANSI
+	normalFaintedColor := aec.NewBuilder(aec.DefaultF).Faint().ANSI
+	greenColor := aec.NewBuilder(aec.GreenF).ANSI
+	if !out.IsTerminal() {
+		headerColor = noColor{}
+		topNameColor = noColor{}
+		normalColor = noColor{}
+		normalFaintedColor = noColor{}
+		greenColor = noColor{}
+	}
+
+	// Print headers
+	for i, h := range headers {
+		if i > 0 {
+			_, _ = fmt.Fprint(out, strings.Repeat(" ", headerSpacing))
+		}
+
+		_, _ = fmt.Fprint(out, h.PrintC(headerColor, h.Title))
+	}
+
+	_, _ = fmt.Fprintln(out)
+
+	printDetails := func(clr aec.ANSI, details imageDetails) {
+		if len(headers) <= 1 {
+			return
+		}
+		truncID := stringid.TruncateID(details.ID)
+		_, _ = fmt.Fprint(out, headers[1].Print(clr, truncID))
+		_, _ = fmt.Fprint(out, strings.Repeat(" ", headerSpacing))
+
+		if len(headers) <= 2 {
+			return
+		}
+		_, _ = fmt.Fprint(out, headers[2].Print(clr, details.DiskUsage))
+		_, _ = fmt.Fprint(out, strings.Repeat(" ", headerSpacing))
+
+		if len(headers) <= 3 {
+			return
+		}
+		if details.Used {
+			_, _ = fmt.Fprint(out, headers[3].Print(greenColor, "✔"))
+		} else {
+			_, _ = fmt.Fprint(out, headers[3].Print(clr, " "))
+		}
+		_, _ = fmt.Fprint(out, strings.Repeat(" ", headerSpacing))
+	}
+
+	// Print images
+	for _, img := range images {
+		_, _ = fmt.Fprintln(out, "")
```

Comment:
This looks a bit weird to me, but it's purely a nit/stylistic – I don't super like the newline between the headers and the image names:
![Screenshot 2024-08-13 at 10 20 58](https://github.com/user-attachments/assets/0df0b3ee-1ab5-4321-afd1-b391b71da6f6)
IMO this looks a bit better:
![Screenshot 2024-08-13 at 10 24 11](https://github.com/user-attachments/assets/4c2ddf65-872f-4cb0-b6e9-b7b689c25317)
Could also have it only between each image "tree" but not before the first one:
![Screenshot 2024-08-13 at 10 25 58](https://github.com/user-attachments/assets/8ab77d55-9617-494a-9a69-565415b040d5)

Reply:
Removed the first newline